### PR TITLE
improving docs

### PR DESCRIPTION
**Pull request by choldgraf**
This is a general improvement to the docs. Updates the theme, adds API auto-linking with sphinx gallery, and uses autodoc to generate API documentation. See a link to what they look like here:

http://predictablynoisy.com/shablona/

LMKWYT

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # containing dir.
 #
 # Note that not all possible configuration values are present in this
-# autogenerated file.
+# autogene`rated file.
```
Comment (arokem) [inline doc/conf.py:11]: Whoops. Typo.

**Review by arokem — APPROVED** [review]
Looks great! Thanks for doing this. 

There's one small typo, but other than that, no comments here.

**Comment by arokem** [comment]
Actually, I am going to go ahead and merge this, and fix that typo myself.

**Comment by choldgraf** [comment]
sweet! thanks man :-)

**Pull request merged**